pipeline/splicing/calcVarPriors.py

```diff
@@ -139,22 +168,314 @@ def getVarType(variant):
+    varStrand = getVarStrand(variant)
+    varOutBounds = varOutsideBoundaries(variant)
+    if varOutBounds == False:
+        for exon in varExons.keys():
+            exonStart = varExons[exon]["exonStart"]
+            exonEnd = varExons[exon]["exonEnd"]
+            if varStrand == "+":
+                if varGenPos > exonStart and varGenPos <= exonEnd:
+                    return True
+            else:
+                if varGenPos <= exonStart and varGenPos >= exonEnd:
+                    return True
+        return False
+    else:
+        return "varOutBounds"    
+
+    
+def varInSpliceDonor(variant):
+    '''
+    Given a variant, determines if a variant is in reference transcript's splice donor region
+    splice donor region = last 3 bases in exon and first 6 bases in intron
+    Returns True if variant in splice donor region
+    '''
+    varGenPos = int(variant["Pos"])
+    varStrand = getVarStrand(variant)
+    donorBounds = getRefSpliceDonorBoundaries(variant)
+    for exon in donorBounds.keys():
+        donorStart = donorBounds[exon]["donorStart"]
+        donorEnd = donorBounds[exon]["donorEnd"]
+        if varStrand == "+":
+            if varGenPos >= donorStart and varGenPos <= donorEnd:
+                return True
+        else:
+            if varGenPos <= donorStart and varGenPos >= donorEnd:
+                return True
+    return False
+
+
+def varInSpliceAcceptor(variant):
+    '''
+    Given a variant, determines if a variant is in reference transcript's splice acceptor region
+    splice acceptor region = 20 bases preceding exon and first 3 bases in exon
+    Returns True if variant in splice acceptor region
+    '''
+    varGenPos = int(variant["Pos"])
+    varStrand = getVarStrand(variant)
+    acceptorBounds = getRefSpliceAcceptorBoundaries(variant)
+    for exon in acceptorBounds.keys():
+        acceptorStart = acceptorBounds[exon]["acceptorStart"]
+        acceptorEnd = acceptorBounds[exon]["acceptorEnd"]
+        if varStrand == "+":
+            if varGenPos >= acceptorStart and varGenPos <= acceptorEnd:
+                return True
+        else:
+            if varGenPos <= acceptorStart and varGenPos >= acceptorEnd:
+                return True
+    return False
+
+
+def varInCiDomain(variant, boundaries):
+    '''
+    Given a variant, determines if variant is in a clinically important domain
+    Second argument determiens which boundaries (ENIGMA or PRIORS) are used for CI domains
+    Returns True if variant in CI domain
+    '''
+    varGenPos = int(variant["Pos"])
+    varGene = variant["Gene_Symbol"]
+    inExon = varInExon(variant)
+    if varGene == "BRCA1":
+        for domain in brca1CIDomains[boundaries].keys():
+            domainStart = brca1CIDomains[boundaries][domain]["domStart"]
+            domainEnd = brca1CIDomains[boundaries][domain]["domEnd"]
+            if varGenPos <= domainStart and varGenPos >= domainEnd and inExon == True:
+                return True
```

Comment:
return in for loop, double check

Reply:
Same principal as above, wanted the code to exit immediately if variant position was in a clinically important domain. Again, can change if it makes more sense to do it a different way

Reply:
I could also potentially use a "checkWithinBoundaries" function here but would have to then have to just add another line to check that the inExon == True before running the checkWithinBoundaries funciton

Reply:
always best to keep it DRY (don't repeat yourself) when possible